File: src/main/java/com/kbank/convenience/stock/stock/mediation/client/ListedStockService.java
package com.kbank.convenience.stock.stock.mediation.client;

import com.kbank.convenience.stock.stock.mediation.client.dto.*;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.cloud.openfeign.SpringQueryMap;
import org.springframework.web.bind.annotation.GetMapping;
import reactivefeign.spring.config.ReactiveFeignClient;
import reactor.core.publisher.Mono;

@ReactiveFeignClient(value = "listed-stock-service", url = "http://127.0.0.1:8088/listed-stock-service")
public interface ListedStockService {
    @GetMapping("/v1/listedStock/{itemCodeNumber}")
    Mono<GetListedStockResponse> getListedStock(@PathVariable("itemCodeNumber") String itemCodeNumber);
    @GetMapping("/v1/listedStock/{itemCodeNumber}/price/latest")
    Mono<GetListedStockLatestPriceResponse> getListedStockLatestPrice(@PathVariable("itemCodeNumber")String itemCodeNumber);
    @GetMapping("/v1/listedStock/{itemCodeNumber}/prices")
    Mono<GetListedStockPricesResponse> getListedStockPrices(@PathVariable("itemCodeNumber") String itemCodeNumber,@SpringQueryMap GetListedStockPricesRequest request);
    @GetMapping("/v1/listedStock/financial/ratio/{itemCodeNumber}")
    Mono<GetListedStockFinancialRatioResponse> getListedStockFinancialRatio(@PathVariable("itemCodeNumber")String itemCodeNumber);
    @GetMapping("/v1/listedStock/financial/statement/latest/{itemCodeNumber}")
    Mono<GetListedStockFinancialStatementResponse> getListedStockFinancialStatement(@PathVariable("itemCodeNumber")String itemCodeNumber);
    @GetMapping("/v1/listedStock/summary/{itemCodeNumber}")
    Mono<GetListedStockSummaryResponse> getListedStockSummary(@PathVariable("itemCodeNumber")String itemCodeNumber);
    @GetMapping("/v1/listedStock/financial/statement/past/{itemCodeNumber}/{targetFinancialStatement}")
    Mono<GetListedStockPastFinancialStatementsResponse> getListedStockPastFinancialStatements(@PathVariable("itemCodeNumber") String itemCodeNumber
            , @PathVariable("targetFinancialStatement") String targetFinancialStatement);
}

File: src/main/java/com/kbank/convenience/stock/stock/mediation/client/ListedStockService.kt
package com.kbank.convenience.stock.stock.mediation.client

import com.kbank.convenience.stock.stock.mediation.client.dto.*
import feign.Param
import feign.QueryMap
import feign.RequestLine


interface ListedStockService {
    @RequestLine("GET /v1/listedStock/{itemCodeNumber}")
    fun getListedStock(@Param("itemCodeNumber") itemCodeNumber: String?): GetListedStockResponse

    @RequestLine("GET /v1/listedStock/{itemCodeNumber}/price/latest")
    fun getListedStockLatestPrice(@Param("itemCodeNumber") itemCodeNumber: String?): GetListedStockLatestPriceResponse

    @RequestLine("GET /v1/listedStock/{itemCodeNumber}/prices")
    fun getListedStockPrices(@Param("itemCodeNumber") itemCodeNumber: String?, @QueryMap request: GetListedStockPricesRequest?): GetListedStockPricesResponse

    @RequestLine("GET /v1/listedStock/financial/ratio/{itemCodeNumber}")
    fun getListedStockFinancialRatio(@Param("itemCodeNumber") itemCodeNumber: String?): GetListedStockFinancialRatioResponse

    @RequestLine("GET /v1/listedStock/financial/statement/latest/{itemCodeNumber}")
    fun getListedStockFinancialStatement(@Param("itemCodeNumber") itemCodeNumber: String?): GetListedStockFinancialStatementResponse

    @RequestLine("GET /v1/listedStock/summary/{itemCodeNumber}")
    fun getListedStockSummary(@Param("itemCodeNumber") itemCodeNumber: String?): GetListedStockSummaryResponse

    @RequestLine("GET /v1/listedStock/financial/statement/past/{itemCodeNumber}/{targetFinancialStatement}")
    fun getListedStockPastFinancialStatements(@Param("itemCodeNumber") itemCodeNumber: String?, @Param("targetFinancialStatement") targetFinancialStatement: String?): GetListedStockPastFinancialStatementsResponse
}
